File: app/src/main/java/com/fyp/agrifarm/ui/DetailsActivity.java
package com.fyp.agrifarm.ui;

import android.content.Intent;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.lifecycle.ViewModelProvider;

import com.fyp.agrifarm.R;
import com.fyp.agrifarm.repo.NewsEntity;
import com.fyp.agrifarm.repo.NewsSharedViewModel;

public class DetailsActivity extends AppCompatActivity {

    public static final String MODE = "newsMode";
    public static final String MODE_NEWS = "newsMode";
    public static final String MODE_YOUTUBE = "newsMode";
    public static final String MODE_PRICE = "newsMode";

    public static final String KEY_TITLE = "title";
    public static final String KEY_DESCRIPTION = "desc";
    public static final String KEY_IMAGE = "image";
    public static final String KEY_DATE = "date";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_detail);

        FragmentManager fm = getSupportFragmentManager();

        Intent intent = getIntent();
        String mode = intent.getStringExtra(MODE);

        switch (mode) {
            case MODE_NEWS:

                String title = intent.getStringExtra(KEY_TITLE);
                String desc = intent.getStringExtra(KEY_DESCRIPTION);
                String image = intent.getStringExtra(KEY_IMAGE);
                String date = intent.getStringExtra(KEY_DATE);
                NewsSharedViewModel newsSharedViewModel = new ViewModelProvider(this).get(NewsSharedViewModel.class);
                newsSharedViewModel.selectNews(new NewsEntity(title, desc, image, date));

                Fragment fragment =
                        fm.findFragmentByTag(NewsDetailsFragment.TAG);
                fragment = (fragment == null) ? new NewsDetailsFragment() : fragment;
                fm.beginTransaction().replace(R.id.container_details_activity, fragment).commitNow();
                break;
        }
    }
}

File: app/src/main/java/com/fyp/agrifarm/app/DetailsActivity.kt
package com.fyp.agrifarm.app

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.ViewModelProvider
import com.fyp.agrifarm.R

import com.fyp.agrifarm.app.news.ui.NewsDetailsFragment
import com.fyp.agrifarm.app.news.viewmodel.NewsSharedViewModel

class DetailsActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_detail)

        val fm = supportFragmentManager
        val intent = intent
        when (intent.getStringExtra(MODE)) {
            MODE_NEWS -> {
                val newsId = intent.getIntExtra(KEY_ID, -1)
                val newsSharedViewModel = ViewModelProvider(this).get(NewsSharedViewModel::class.java)
                newsSharedViewModel.selectNews(newsId)

                var fragment = fm.findFragmentByTag(NewsDetailsFragment.TAG)
                fragment = fragment ?: NewsDetailsFragment()
                fragTransaction(fragment).commit()
            }
        }
    }

    companion object {
        const val MODE = "mode"
        const val MODE_NEWS = "newsMode"
        const val MODE_YOUTUBE = "youtubeMode"
        const val MODE_PRICE = "priceMode"
        const val KEY_TITLE = "title"
        const val KEY_DESCRIPTION = "desc"
        const val KEY_IMAGE = "image"
        const val KEY_ID = "newsId"
    }
}